Documentation/ExtensionArchitecture/Extbase/Reference/FileUpload.rst

```diff
@@ -0,0 +1,554 @@
+*   delete the affected :php:`sys_file_reference` record
+*   delete the affected file
+
+Internally, Extbase uses :php:`FileUploadDeletionConfiguration` objects to track
+file deletions for properties of arguments. Files are deleted directly without
+checking whether the current file is referenced by other objects.
+
+Apart from using this ViewHelper, it is of course still possible to manipulate
+:php-short:`\TYPO3\CMS\Core\Resource\FileReference` properties with custom logic before persistence.
+
+..  _extbase_fileupload_attribute-psr-event:
+
+ModifyUploadedFileTargetFilenameEvent
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :php-short:`\TYPO3\CMS\Extbase\Event\Service\ModifyUploadedFileTargetFilenameEvent`
+allows event listeners to
+alter a filename of an uploaded file before it is persisted.
+
+Event listeners can use the method `getTargetFilename()` to retrieve the filename
+used for persistence of a configured uploaded file. The filename can then be
+adjusted via `setTargetFilename()`. The relevant configuration can be retrieved
+via `getConfiguration()`.
+
+Multi-step form handling
+------------------------
+
+The implementation of the file upload feature in Extbase intentionally requires to
+handle the FileUpload directly within the validation/persistence step of a controller
+action.
+
+If you have a multi-step process in place, where the final persistence of a domain model
+object is only performed later on, you will need to deal with the created files.
+
+For example, you may want to implement a preview before the final domain model
+entity is persisted.
+
+Some possible ways to deal with this:
+
+*   Implement the file handling as a :ref:`DTO <concept-dto>`. The key idea here is
+    to decouple the uploaded file into its own domain model object. You can pass that
+    along (including its persistence identity) from one form step to the next, and only
+    in the final step you would take care of transferring the data of this DTO into
+    your actual domain model, and attach the FileReference object.
+
+*   Or use client-side JavaScript. You could create a stub in your Fluid template that
+    has placeholders for user-specified data, and then fills the actual data (before
+    the form is submitted) into these placeholders. You can use the JavaScript :js:`FileReader()`
+    object to access and render uploaded files.
+
+*   Or pass along the created :php:`FileReference` object from the action that processes
+    the actual file upload data in a session variable or a hidden entity, so that you can
+    later access that before the final domain model persistence action. Note however that
+    requires you to also handle stale/temporary files manually, and manage possible outdated
+    :sql:`sys_file` records. In this case you might better use the regular PHP `UploadedFile`
+    handling to manage file uploads, as you are bypassing key concepts of the extbase file
+    upload handling feature.
```

Comment:
Lets remove this part, since we do not have a FileReference at this point, since a FileReference is always referenced to a sys_file and a foreign table.

Showing the sys_file_reference.uid in a hidden field in Frontend is bad in terms of security, since a user can modify it and thereby change the reference to every sys_file_reference.